OpenGLTechTalks/Basic/src/Shader.cpp

```diff
@@ -0,0 +1,128 @@
+#include "Shader.h"
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <ostream>
+
+namespace core
+{
+    Shader::Shader() : m_ID{}
+    {
+    }
+
+    Shader::Shader(const char* vertexPath, const char* fragmentPath) : Shader()
+    {
+        // 1. retrieve the vertex/fragment source code from filePath
+        std::string vertexCode;
+        std::string fragmentCode;
+        std::ifstream vShaderFile;
+        std::ifstream fShaderFile;
+
+        vShaderFile.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+        fShaderFile.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+
+        try
+        {
+            vShaderFile.open(vertexPath);
+            fShaderFile.open(fragmentPath);
+            std::stringstream vShaderStream, fShaderStream;
+            vShaderStream << vShaderFile.rdbuf();
+            fShaderStream << fShaderFile.rdbuf();
+            vShaderFile.close();
+            fShaderFile.close();
+            vertexCode = vShaderStream.str();
+            fragmentCode = fShaderStream.str();
+        }
+        catch (const std::ifstream::failure&)
+        {
+            std::cout << "ERROR::SHADER::FILE_NOT_SUCCESSFULLY_READ::" << std::endl;
+        }
+
+        // 2. compile shaders
+        const char* vShaderCode = vertexCode.c_str();
+        const char* fShaderCode = fragmentCode.c_str();
+
+        unsigned int vertex, fragment;
```

Comment:
better to use GLuint. It will showing us that that part of code is connected to OpenGL directly